src/marqo/s2_inference/processing/image.py

```diff
@@ -211,9 +211,11 @@ def _load_and_cache_model(self):
 
             self.model, self.preprocess = func(self.model_name, self.device)
 
-            available_models[model_cache_key] = self.model, self.preprocess
+            available_models[model_cache_key] = {"model": (self.model, self.preprocess),
+                                                 "most_recently_used_time" : datetime.datetime.now()}
```

Comment:
nit, important keys are probably better declared by an enum:

```
class ModelCacheObjects:
    model = "model"
    most_recently_used_time = "most_recently_used_time"
```

This prevents hard to detect bugs (like from a typo)